ols/src/llms/llm_loader.py

```diff
@@ -33,86 +33,42 @@ class ModelConfigMissingError(LLMConfigurationError):
     """No configuration exists for the requested model name."""
 
 
-class ModelConfigInvalidError(LLMConfigurationError):
-    """Model configuration is not valid."""
+def _resolve_provider_config(provider, model, providers_config) -> ProviderConfig:
+    """Ensure the provided inputs (provider/model) are valid in config.
 
+    Return respective provider configuration.
+    """
+    if provider not in providers_config.providers:
+        raise UnknownProviderError(
+            f"Provider '{provider}' is not a valid provider. "
+            f"Valid providers are: {list(providers_config.providers.keys())}"
+        )
 
-class LLMLoader:
-    """Load LLM backend.
+    provider_config = providers_config.providers.get(provider)
 
-    Example:
-        ```python
-        # using the class and overriding specific parameters
-        llm_params = {'temperature': 0.02, 'top_p': 0.95}
+    if model not in provider_config.models:
+        raise ModelConfigMissingError(
+            f"Model '{model}' is not a valid model for provider '{provider}'. "
+            f"Valid models are: {list(provider_config.models.keys())}"
+        )
 
-        bare_llm = LLMLoader(provider="openai", model="gpt-3.5-turbo", llm_params=llm_params).llm
-        llm_chain = LLMChain(llm=bare_llm, prompt=prompt)
-        ```
-    """
+    return provider_config
+
+
+def load_llm(provider: str, model: str, llm_params: Optional[dict]) -> LLM:
```

Comment:
Replaced LLMLoader with func (needs polishing). WDYT?

Reply:
it's IMHO much better solution